kedro-datasets/kedro_datasets/spark/__init__.py

```diff
@@ -19,5 +20,6 @@
         "spark_hive_dataset": ["SparkHiveDataset"],
         "spark_jdbc_dataset": ["SparkJDBCDataset"],
         "spark_streaming_dataset": ["SparkStreamingDataset"],
+        "spark_gbq_dataset": ["GBQQueryDataset"],
```

Comment:
To be consistent, this should be

```suggestion
        "gbq_dataset": ["GBQQueryDataset"],
```

(see `spark.deltatable_dataset`, `pandas.gbq_dataset`)

(and move it right after `deltatable_dataset`, while you're at it)